esp-hal/src/dma/mod.rs

```diff
@@ -2155,12 +2302,7 @@ impl DmaTxBuf {
 impl DmaTxBuffer for DmaTxBuf {
     fn prepare(&mut self) -> Preparation {
         for desc in self.descriptors.iter_mut() {
-            // Give ownership to the DMA
-            desc.set_owner(Owner::Dma);
-
-            if desc.next.is_null() {
-                break;
-            }
+            desc.reset_for_tx(false);
```

Comment:
I was hoping to add an optimization here to skip this for loop if the first descriptor's owner bit was already set to DMA.
I'm not sure how sensible it'll look now that the "reset" logic is behind a method.

Reply:
Sounds a bit like wasted CPU cycles to me, but also nothing prevents you to add that if it isn't.